### PR TITLE
OCM-3298: Add BYOVPC module

### PR DESCRIPTION
**Pull request by nirarg**

### CODE REVIEW

File: examples/rosa-classic-with-unmanaged-oidc/main.tf [hunk 1]
```diff
@@ -35,11 +35,11 @@ module "operator_roles" {
   oidc_endpoint_url    = module.oidc_provider.oidc_endpoint_url
 }
 
-// module "vpc_private" {
-//   source = "../../modules/vpc-private"
-// 
-//   name = "${var.cluster_name}-vpc"
-// }
+module "vpc" {
```
Comment (bardielle) [inline examples/rosa-classic-with-unmanaged-oidc/main.tf:38]: I think the example name should include BYO VPC

File: modules/account-iam-resources/main.tf [hunk 2]
```diff
@@ -107,17 +107,10 @@ locals {
...
-      "production"  = "710019948333",
-      "staging"     = "644306948063",
-      "integration" = "896164604406",
-      "local"       = "765374464689"
-    },
-    var.ocm_environment, "710019948333"
-  )
 }
 
+data "rhcs_info" "current" {}
```
Comment (bardielle) [inline modules/account-iam-resources/main.tf:112]: :+1:

File: modules/account-iam-resources/versions.tf [hunk 3]
```diff
@@ -7,7 +7,7 @@ terraform {
       version = ">= 4.0"
     }
     rhcs = {
-      version = ">= 1.1.0"
+      version = "1.4.0-prerelease.3"
```
Comment (bardielle) [inline modules/account-iam-resources/versions.tf:10]: I guess it should be 1.4.0

File: modules/vpc/main.tf [hunk 4]
```diff
@@ -0,0 +1,196 @@
+locals {
+  azs                = slice(data.aws_availability_zones.available.names, 0, 3)
+  private_cidr_range = [for k, v in local.azs : cidrsubnet(var.vpc_cidr, 8, k)]
```
Comment (bardielle) [inline modules/vpc/main.tf:3]: can `var.vpc_cidr` be from an output from datasource ?

File: modules/vpc/main.tf [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+  route_table_ids = concat(
+    aws_route_table.private_routes.*.id,
+    aws_route_table.default.*.id,
+  )
+
+  tags = var.tags
+}
+
+resource "aws_vpc_dhcp_options" "main" {
+  domain_name         = data.aws_region.current.name == "us-east-1" ? "ec2.internal" : format("%s.compute.internal", data.aws_region.current.name)
```
Comment (bardielle) [inline modules/vpc/main.tf:32]: Why not always use the format?

File: modules/vpc/main.tf [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+  route_table_id = aws_route_table.default.id
+}
+
+resource "aws_route" "igw_route" {
+  destination_cidr_block = "0.0.0.0/0"
+  route_table_id         = aws_route_table.default.id
+  gateway_id             = aws_internet_gateway.igw.id
+
+  timeouts {
+    create = "20m"
```
Comment (bardielle) [inline modules/vpc/main.tf:76]: maybe it can be a variable with a default value

File: modules/account-iam-resources/versions.tf (hunk 3 above)
Comment (nirarg) [reply modules/account-iam-resources/versions.tf:10]: I will change it to 1.4.0 in different commit after release

File: examples/rosa-classic-with-unmanaged-oidc/main.tf (hunk 1 above)
Comment (nirarg) [reply examples/rosa-classic-with-unmanaged-oidc/main.tf:38]: Lets push this in another PR

File: modules/vpc/main.tf (hunk 5 above)
Comment (nirarg) [reply modules/vpc/main.tf:32]: Lets push this in another PR

File: modules/vpc/main.tf (hunk 6 above)
Comment (nirarg) [reply modules/vpc/main.tf:76]: Lets push this in another PR

**Pull request merged**